### ISSUE TITLE
Does not work on IE and Microsoft Edge

### ISSUE THREAD

**Issue by azharvalley** [shahzam/DialogTriggerJS#3]
This is not working on Microsoft Edge and IE. Can you help me with this @shahzam

**Comment by shahzam** [comment]
Hi, sorry for the delay. Was there a particular trigger type that was causing trouble in IE? Also, can you share a snippet of code showing how you're using it?

**Comment by azharvalley** [comment]
I am using trigger: 'exitIntent', 
----------------
When lmrPopup() function is called I check cookie name "lowermyratespopped" should be empty. If the cookie is empty, then openpopup() function is called which opens the modal pop up. 
Here is my snippet of code. 

            function lmrPopup() {
                if (get_cookie('lowermyratespopped') == ' ' ) {
                    openpopup()
                }
            }

            // A trigger based on "exit intent" when the cursor is detected to go above the top of the browser window (useful in desktop scenarios)
            var dtExit = new DialogTrigger(lmrPopup, { trigger: 'exitIntent' });

**Comment by shahzam** [comment]
Hi, I was able to test the code out on IE11 and it seems to work okay on my end. Does your code snippet work for you as expected under Chrome or another browser?

**Comment by azharvalley** [comment]
Yes, @shahzam my code snippet works for Chrome. Can you share some sample which you used for IE11

**Comment by shahzam** [comment]
Hi @azharvalley, sorry for the delay. Here is the full HTML file with Javascript that I tested in IE11. It will show an alert when the cursor is moved out of the window for the first time, to demonstrate the bare-bones functionality. Let me know if this works for you.

```
<!DOCTYPE html>
<html>
<head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>

<script>
function DialogTrigger(callback, options) {
  // Becomes this.options
  var defaults = {
    trigger: 'timeout',
    target: '',
    timeout: 0,
    percentDown: 50, // Used for 'percent' to define a down scroll threshold of significance (based on page height)
    percentUp: 10, // Used for 'percent' to define a threshold of upward scroll after down threshold is reached
    scrollInterval: 1000 // Frequency (in ms) to check for scroll changes (to avoid bogging the UI)
  }

  this.complete = false; // Let's us know if the popup has been triggered

  this.callback = callback;
  this.timer = null;
  this.interval = null;

  this.options = jQuery.extend(defaults, options);

  this.init = function() {
    if (this.options.trigger == 'exitIntent' || this.options.trigger == 'exit_intent') {
      var parentThis = this;

      jQuery(document).on('mouseleave', function(e) {
        //console.log(e.clientX + ',' + e.clientY); // IE returns negative values on all sides

        if (!parentThis.complete && e.clientY < 0) { // Check if the cursor went above the top of the browser window
          parentThis.callback();
          parentThis.complete = true;
          jQuery(document).off('mouseleave');
        }
      });

    } else if (this.options.trigger == 'target') {
      if (this.options.target !== '') {
        // Make sure the target exists
        if (jQuery(this.options.target).length == 0) {
          this.complete = true;
        } else {
          var targetScroll = jQuery(this.options.target).offset().top;

          var parentThis = this;

          // Only check the scroll position every few seconds, to avoid bogging the UI
          this.interval = setInterval(function() {
            if (jQuery(window).scrollTop() >= targetScroll) {
              clearInterval(parentThis.interval);
              parentThis.interval = null;

              if (!parentThis.complete) {
                parentThis.callback();
                parentThis.complete = true;
              }
            }
          }, this.options.scrollInterval);
        }
      }

    } else if (this.options.trigger == 'scrollDown') {
      // Let the user scroll down by some significant amount
      var scrollStart = jQuery(window).scrollTop();
      var pageHeight = jQuery(document).height();

      var parentThis = this;

      if (pageHeight > 0) {
        // Only check the scroll position every few seconds, to avoid bogging the UI
        this.interval = setInterval(function() {
          var scrollAmount = jQuery(window).scrollTop() - scrollStart;
          if (scrollAmount < 0) {
            scrollAmount = 0;
            scrollStart = jQuery(window).scrollTop();
          }
          var downScrollPercent = parseFloat(scrollAmount) / parseFloat(pageHeight);

          if (downScrollPercent > parseFloat(parentThis.options.percentDown) / 100) {
            clearInterval(parentThis.interval);
            parentThis.interval = null;

            if (!parentThis.complete) {
              parentThis.callback();
              parentThis.complete = true;
            }
          }

        }, this.options.scrollInterval);
      }

    } else if (this.options.trigger == 'scrollUp') {
      // Let the user scroll down by some significant amount
      var scrollStart = jQuery(window).scrollTop();
      var pageHeight = jQuery(document).height();

      var parentThis = this;

      if (pageHeight > 0) {
        // Only check the scroll position every few seconds, to avoid bogging the UI
        this.interval = setInterval(function() {
          var scrollAmount = scrollStart - jQuery(window).scrollTop();
          if (scrollAmount < 0) {
            scrollAmount = 0;
            scrollStart = jQuery(window).scrollTop();
          }
          var upScrollPercent = parseFloat(scrollAmount) / parseFloat(pageHeight);

          if (upScrollPercent > parseFloat(parentThis.options.percentUp) / 100) {
            clearInterval(parentThis.interval);
            parentThis.interval = null;

            if (!parentThis.complete) {
              parentThis.callback();
              parentThis.complete = true;
            }
          }

        }, this.options.scrollInterval);
      }

    } else if (this.options.trigger == 'timeout') {
      this.timer = setTimeout(this.callback, this.options.timeout);
    }

  };

  this.cancel = function() {
    if (this.timer !== null) {
      clearTimeout(this.timer);
      this.timer = null;
    }

    if (this.interval !== null) {
      clearInterval(this.interval);
      this.interval = null;
    }

    this.complete = true;
  }

  this.init();
}

function lmrPopup() {
  //if (get_cookie('lowermyratespopped') == ' ' ) {
  alert('hello')
  //}
}

// A trigger based on "exit intent" when the cursor is detected to go above the top of the browser window (useful in desktop scenarios)
var dtExit = new DialogTrigger(lmrPopup, {
  trigger: 'exitIntent'
});

</script>
</head>

<body>

<h1>Exit Intent Test</h1>

<p id="demo">Try moving the cursor out of the window.</p>

</body>
</html> 
```

**Comment by azharvalley** [comment]
Hi @shahzam thanks for your reply. I too checked this after a very long time. Though this code did not worked for me on IE and edge. But when I did a small change it did worked on IE and Edge as well.

I am adding my comment so that it might help someone who is struggling on IE and Edge. 


I found that Internet Explorer and Edge does not return negative values of the co-ordinates when cursor is moved out of the current window tab. The "Y" value returns up to 0 in IE and Edge when cursor moved outside. No negative value is returned due to this the pop up did not triggered on these two browsers (IE and Edge). 

So instead of checking "Y" co-ordinates to negative value, I am checking if "Y" co-ordinates is less than or equal to "1" and finally it worked on IE and Edge as well. 

Here is one line of change I did in the JS file. 

``` 
if (!parentThis.complete && e.clientY <= 1) { // Check if the cursor went above the top of the browser window
